service/matching/taskListManager.go

```diff
@@ -949,3 +969,16 @@ func (c *taskContext) completeTask(err error) {
 func createServiceBusyError(msg string) *s.ServiceBusyError {
 	return &s.ServiceBusyError{Message: msg}
 }
+
+func (c *taskListManagerImpl) updateLastAddTaskTime() {
+	c.lastAddTaskTimeLock.Lock()
```

Comment:
I think we should do this without adding another lock.

Reply:
Update with better solution in getTaskPump